### DISCUSSION TITLE
Same error showing in Netlify deploy and Github pages CI/CD

### DISCUSSION THREAD

**Discussion by sachinshilwal** [vuepress/core#836] [Q&A] [answered]
Hi So I am getting error while running `npm install && npm run docs:build` like this:

```
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for esbuild-windows-64@0.14.36: wanted {"os":"win32","arch":"x64"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    win32
npm ERR! notsup Valid Arch:  x64
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2022-04-18T08_28_06_497Z-debug-0.log
Error: Process completed with exit code 1.
```
Some one with prior knowledge in these help would be appreciated.

**Comment by Mister-Hope** [comment, marked as answer]
if you are a native English speaker or you know English well, read the log and the answer is obvious. You are installing esbuild window edition locally, while your ci is using Linux.

Try removing lock file when committing to ci, or let ci ignore lock file when installing. 

First solution requires knowledge with git
Second solution requires knowledge with npm

**Reply by sachinshilwal** [reply]
yes Just simple googling gave the answer to remove package-lock.json file. I did that and error remained the same. So I came here to ask. 
removing lock file did not solved the problem

**Reply by Mister-Hope** [reply]
Provide enough information before requesting help. I can tell you that vuepress is requiring esbuild instead of `esbuild-windows-64`, so npm is expected to choose the proper version and install for you, and you should not meet this issue.

Visit https://stackblitz.com/fork/vuepress and download it. If uploading this demo still did not work in your ci, fell free to provide ci logs and comment here.

If not, try to make sure if you do not do something wrong, I have no instests guessing the reason, show me a reproduction repo then shall I take a look.

**Reply by Mister-Hope** [reply]
Also, I hope there is no `esbuild-windows-64` in your package.json, or you are obviously joking everyone and wasting others time.

**Reply by sachinshilwal** [reply]
thanks
